src/app/components/category-products/category-products.component.ts

```diff
@@ -25,10 +40,141 @@ export class CategoryProductsComponent {
         this._categoriesService.getOneCategory(data['categoryId'])
       )
     );
+  readonly paginationData$: Observable<{
+    pageSize: number;
+    pageNumber: number;
+  }> = this._activatedRoute.queryParams.pipe(
+    map((data) => {
+      return {
+        pageSize: data['pageSize'] === undefined ? 5 : +data['pageSize'],
+        pageNumber: data['pageNumber'] === undefined ? 1 : +data['pageNumber'],
+      };
+    })
+  );
+  readonly sortedProducts$: Observable<ProductsWithCategoryNameQueryModel[]> =
+    combineLatest([
+      this._activatedRoute.params,
+      this._productsService.getAll(),
+      this._categoriesService.getAllCategory(),
+      this.form.valueChanges.pipe(
+        startWith({ selectFilter: { id: 1, value: 'Featured', order: 'desc' } })
+      ),
+    ]).pipe(
+      map(([params, products, categories, selectFilter]) =>
+        this._mapToProductsWithCategoryName(products, categories)
+          .filter((p) => p.category.id.includes(params['categoryId']))
+          .sort((a, b) => {
+            const select = selectFilter['selectFilter'];
+            if (select.id === 2) {
```

Comment:
you can work out a better query model, because the way it works is sort of magic. If the order of options change, your code will start working. Think of a better solution here